### PR TITLE
Add proof rollup

### PR DESCRIPTION
**Pull request by chrisnojima**
This is a rebase of https://github.com/keybase/client/pull/3693 plus

- [x] Fixes dupes in proofs (new resetProofs action)
- [x] Hooks up the retry menu option

@keybase/react-hackers

### CODE REVIEW

File: shared/profile/render.desktop.js [hunk 1]
```diff
@@ -113,11 +113,11 @@ class Render extends Component<void, Props, State> {
             borderBottom: `1px solid ${globalColors.black_05}`,
           }}
         >
-          <PlatformIcon platform={proof.type} overlay='iconfont-proof-good' overlayColor={globalColors.blue} size={48} />
+          <PlatformIcon platform={proof.type} overlay='icon-proof-success' overlayColor={globalColors.blue} size={48} />
           <Text type='Body' style={{textAlign: 'center', color: globalColors.black_40}}>Posted on<br />{moment(proof.mTime).format('ddd MMM D, YYYY')}</Text>
```
Comment (awendland) [inline shared/profile/render.desktop.js]: We should standardize this date format somewhere. Maybe in `styles/style-guide`?

Comment (awendland) [inline shared/profile/render.desktop.js]: We should standardize this date format somewhere. Maybe in `styles/style-guide`?

**Comment by awendland** [comment]
Trying to review but running into this error at runtime after typing in a Twitter username 
screen shot 2016-08-10 at 2 09 50 pm
screen shot 2016-08-10 at 2 09 29 pm

**Comment by cjb** [comment]
Ahh that's gonna be a can of worms.

Any idea how you managed to have a locked keychain?  The app should unlock it at startup.

**Comment by cjb** [comment]
Oh, like maybe you don't have Save to Keychain on, and you restarted your service after launching the GUI, that would do it.  I don't think we expect users to ever do that.

**Comment by awendland** [comment]
@cjb Thanks for the prognosis. I ran `keybase unlock` and saved to keychain to fix this for the future.

**Comment by awendland** [comment]
Testing:
- [ ] Twitter
- [ ] Reddit
- [ ] GitHub
- [ ] Coinbase
- ~~[ ] HackerNews~~
- [ ] Bitcoin address

**Comment by cjb** [comment]
@awendland welcome, it's a nasty bug that we can't just have a global engine handler for secretUi.getPassphrase and spawn pinentry windows when we get it, but I think it's really hard to fix

**Comment by awendland** [comment]
@chrisnojima Is it desired behaviour that after entering my username into Reddit, then cancelling on the post proof screen, that I now have an unreachable proof for Reddit?

**Comment by chrisnojima** [comment]
@awendland yes. as soon as you enter you username we generate the proof and add it to the sigchain. this has to be how it works currently due to a lot of issues and is totally expected

**Comment by chrisnojima** [comment]
i don't think you need to test each individual proof type, this is just a sanity check

**Comment by awendland** [comment]
@chrisnojima Just finished. Wanted to make certain of the sanity.

**Comment by awendland** [comment]
Last update: after clicking check proof button after posting a proof, then clicking cancel while it is waiting, the app properly returns to the My Profile screen, where it stays momentarily until it jumps to the Proof Verified screen when the check proof completes. It appears that cancel is not actually canceling the RPC?

**Comment by awendland** [comment]
👍 General functionality seems good. Sanity established.

**Comment by chrisnojima** [comment]
and now i have to rebase again :)

**Comment by chrisnojima** [comment]
yeah there can be a timing issue w/ cancel for sure. i'll just ticket

**Pull request merged**